triple-store/triple-store-impl-blazegraph/src/main/java/com/powsybl/triplestore/impl/blazegraph/TripleStoreBlazegraph.java

```diff
@@ -247,60 +249,76 @@ public PropertyBags query(String query) {
     }
 
     @Override
-    public void add(String graph, String objType, PropertyBags statements) {
+    public void add(String contextName, String objNs, String objType, PropertyBags statements) {
         RepositoryConnection cnx = null;
         try {
             cnx = repo.getConnection();
-            addStatements(cnx, graph, objType, statements);
+            addStatements(cnx, contextName, objNs, objType, statements);
 
         } catch (RepositoryException x) {
-            throw new TripleStoreException(String.format("Adding statements for graph %s", graph), x);
+            throw new TripleStoreException(String.format("Adding statements for graph %s", contextName), x);
         } finally {
             if (cnx != null) {
                 try {
                     cnx.close();
                 } catch (RepositoryException x) {
-                    LOG.error("Adding statements for graph {}. Closing repository connection", graph);
+                    LOG.error("Adding statements for graph {}. Closing repository connection", contextName);
                 }
             }
         }
     }
 
-    private void addStatements(RepositoryConnection cnx, String graph, String objType, PropertyBags statements)
-            throws RepositoryException {
-
-        cnx.begin();
-
-        String name = null;
-        RepositoryResult<Resource> ctxs = cnx.getContextIDs();
-        while (ctxs.hasNext()) {
-            String ctx = ctxs.next().stringValue();
-            if (ctx.contains("EQ")) {
-                name = ctx.replace("EQ", graph);
-                break;
+    @Override
+    public String add(String contextName, String objNs, String objType, PropertyBag properties) {
+        RepositoryConnection cnx = null;
+        try {
+            cnx = repo.getConnection();
+            return addStatement(cnx, contextName, objNs, objType, properties);
+        } catch (RepositoryException x) {
+            throw new TripleStoreException(String.format("Adding statements for graph %s", contextName), x);
+        } finally {
+            if (cnx != null) {
+                try {
+                    cnx.close();
+                } catch (RepositoryException x) {
+                    LOG.error("Adding statements for graph {}. Closing repository connection", contextName);
+                }
             }
```

Comment:
Maybe we should move this block in a dedicated function to improve readability:
```
void closeConnection(RepositoryConnection cnx) {
            if (cnx != null) {
                try {
                    cnx.close();
                } catch (RepositoryException x) {
                    LOG.error("Adding statements for graph {}. Closing repository connection", contextName);
                }
            }
}
```

Reply:
Method added